indexer/services/vulcan/src/handlers/order-remove-handler.ts

```diff
@@ -276,16 +277,23 @@ export class OrderRemoveHandler extends Handler {
 
     // If the remaining amount of the order in state is <= 0, the order is filled and
     // does not need to have it's status updated
+    let canceledOrder: OrderFromDatabase | undefined;
     if (stateRemainingQuantums.gt(0)) {
-      await runFuncWithTimingStat(
+      canceledOrder = await runFuncWithTimingStat(
         this.cancelOrderInPostgres(orderRemove),
         this.generateTimingStatsOptions('cancel_order_in_postgres'),
       );
+    } else {
+      canceledOrder = await runFuncWithTimingStat(
+        OrderTable.findById(OrderTable.orderIdToUuid(orderRemove.removedOrderId!)),
+        this.generateTimingStatsOptions('find_order'),
+      );
```

Comment:
Suggestion: This could probably be optimized by not querying for the order if there isn't an entry in the state filled quantums cache for the order. That would mean no fills for the order, and as it would be a short-term order that means it won't be in the DB.